### PR TITLE
Drop Python 3.6 in batch-cli and upgrade Python runtimes

### PR DESCRIPTION
**Pull request by demartinofra**
### Description of changes
* Add tests for Python 3.10 in aws-parallelcluster package
* move support for Python 3.6 and add Python 3.10 tests in aws-paralelcluster-batch-cli
* Bump Python and nodejs versions in API Dockerfile and infra template
* Bump Python version for Lambda functions in cluster templates

### Tests
* Unit tests
* Locally built API docker container

### References
* https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 CHANGELOG
 =========
 
+3.x.x
+-----
+
+**CHANGES**
+- Remove support for Python 3.6 in aws-parallelcluster-batch-cli.
+- Upgrade Python version in API infrastructure, API Docker container and cluster Lambda resources.
```
Comment (lukeseawalker) [inline CHANGELOG.md]: is it worth mentioning nodejs version upgrade?

File: awsbatch-cli/tox.ini [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [tox]
 toxworkdir=../.tox
 envlist =
-    py{36,37,38,39}-{cov,nocov}
+    py{36,37,38,39,310}-{cov,nocov}
```
Comment (lukeseawalker) [inline awsbatch-cli/tox.ini]: remove py36?

File: cli/tox.ini [hunk 3]
```diff
@@ -1,7 +1,7 @@
 [tox]
 toxworkdir=../.tox
 envlist =
-    py{36,37,38,39}-{cov,nocov}
+    py{36,37,38,39,310}-{cov,nocov}
```
Comment (lukeseawalker) [inline cli/tox.ini]: same here for py36

File: CHANGELOG.md (hunk 1 above)
Comment (demartinofra) [reply CHANGELOG.md]: done

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**